breathecode/marketing/tests/urls/tests_lead_id.py

```diff
@@ -0,0 +1,266 @@
+"""
+Test /academy/lead/id
+"""
+from django.utils import timezone
+from datetime import timedelta
+import re, string
+from random import choice, choices, randint
+from mixer.main import Mixer
+from unittest.mock import patch
+from django.urls.base import reverse_lazy
+from rest_framework import status
+from breathecode.tests.mocks import (
+    GOOGLE_CLOUD_PATH,
+    apply_google_cloud_client_mock,
+    apply_google_cloud_bucket_mock,
+    apply_google_cloud_blob_mock,
+)
+from ..mixins import MarketingTestCase
+
+
+def random_string():
+    return ''.join(choices(string.ascii_letters, k=10))
+
+
+def generate_form_entry_kwargs():
+    """That random values is too long that i prefer have it in one function"""
+    return {
+        'fb_leadgen_id': randint(0, 9999),
+        'fb_page_id': randint(0, 9999),
+        'fb_form_id': randint(0, 9999),
+        'fb_adgroup_id': randint(0, 9999),
+        'fb_ad_id': randint(0, 9999),
+        'gclid': random_string(),
+        'first_name': choice(['Rene', 'Albert', 'Immanuel']),
+        'last_name': choice(['Descartes', 'Camus', 'Kant']),
+        'email': choice(['a@a.com', 'b@b.com', 'c@c.com']),
+        'phone': choice(['123', '456', '789']),
+        'course': random_string(),
+        'client_comments': random_string(),
+        'location': random_string(),
+        'language': random_string(),
+        'utm_url': random_string(),
+        'utm_medium': random_string(),
+        'utm_campaign': random_string(),
+        'utm_source': random_string(),
+        'referral_key': random_string(),
+        'gclid': random_string(),
+        'tags': random_string(),
+        'automations': random_string(),
+        'street_address': random_string(),
+        'country': random_string(),
+        'city': random_string(),
+        'latitude': randint(0, 9999),
+        'longitude': randint(0, 9999),
+        'state': random_string(),
+        'zip_code': randint(0, 9999),
+        'browser_lang': random_string(),
+        'storage_status': choice(['PENDING', 'PERSISTED']),
+        'lead_type': choice(['STRONG', 'SOFT', 'DISCOVERY']),
+        'deal_status': choice(['WON', 'LOST']),
+        'sentiment': choice(['GOOD', 'BAD']),
+    }
+
+
+class CohortLeadIdSuite(MarketingTestCase):
+    """Test /academy/lead"""
+    """
+    🔽🔽🔽 Single lead with data wrong id
+    """
+
+    @patch(GOOGLE_CLOUD_PATH['client'], apply_google_cloud_client_mock())
+    @patch(GOOGLE_CLOUD_PATH['bucket'], apply_google_cloud_bucket_mock())
+    @patch(GOOGLE_CLOUD_PATH['blob'], apply_google_cloud_blob_mock())
+    def test_lead_wrong_id(self):
+        """Test /lead/:id/ with data wrong id"""
+        self.headers(academy=1)
+        url = reverse_lazy('marketing:academy_lead_id', kwargs={'lead_id': 2})
+        model = self.generate_models(
+            authenticate=True,
+            profile_academy=True,
+            capability='read_lead',
+            role='potato',
+            form_entry=True,
```

Comment:
remove the `form_entry=True` and change `wargs={'lead_id': 2}` to `wargs={'lead_id': 1}`